src/Symfony/Component/Cache/Adapter/RedisTagAwareAdapter.php

```diff
@@ -62,26 +56,18 @@ class RedisTagAwareAdapter extends AbstractTagAwareAdapter
      */
     private const DEFAULT_CACHE_TTL = 8640000;
 
-    /**
-     * @var bool|null
-     */
-    private $redisServerSupportSPOP = null;
-
     /**
      * @param \Redis|\RedisArray|\RedisCluster|\Predis\ClientInterface $redisClient     The redis client
      * @param string                                                   $namespace       The default namespace
      * @param int                                                      $defaultLifetime The default lifetime
-     *
-     * @throws \Symfony\Component\Cache\Exception\LogicException If phpredis with version lower than 3.1.3.
      */
     public function __construct($redisClient, string $namespace = '', int $defaultLifetime = 0, MarshallerInterface $marshaller = null)
     {
-        $this->init($redisClient, $namespace, $defaultLifetime, $marshaller);
-
-        // Make sure php-redis is 3.1.3 or higher configured for Redis classes
-        if (!$this->redis instanceof \Predis\ClientInterface && version_compare(phpversion('redis'), '3.1.3', '<')) {
-            throw new LogicException('RedisTagAwareAdapter requires php-redis 3.1.3 or higher, alternatively use predis/predis');
+        if ($redisClient instanceof \Predis\ClientInterface && $redisClient->getConnection() instanceof ClusterInterface && !$redisClient->getConnection() instanceof PredisCluster) {
```

Comment:
I don't understand this condition. Why are we supporting PredisCluster, which does client-side clustering, and not RedisCluster which relies on the actual redis-cluster clustering ?

Reply:
Because `rename` on `RedisCluser` is not supported by Predis.